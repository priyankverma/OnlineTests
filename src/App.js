import "./App.css";
import React, { Component } from "react";
import { createStore, applyMiddleware, compose } from "redux";
import { Provider } from "react-redux";
import combinedReducer from "./../src/redux/combinedReducers";

import {
  BrowserRouter as Router,
  Switch,
  Route,
  withRouter,
  Redirect,
} from "react-router-dom";
import Dashboard from "../src/screens/dashboard/dashboard";
import LoginScreen from "./screens/login/loginScreen";
import Summary from "./screens/summary/summary";
import PrivateRoute from "./commonComponents/privateRoute/privateRoute";
const ReduxThunk = require("redux-thunk").default;

let composeEnhancer = compose;
export const store = createStore(
  //creates an instance of store to be used globally
  combinedReducer, // pass in the combined reducers
  composeEnhancer(applyMiddleware(ReduxThunk)) // use the middleware for async actions
);

class App extends Component {
  constructor(props) {
    super(props);

    // Store the previous pathname and search strings
    this.currentPathname = null;
    this.currentSearch = null;
  }
  componentDidMount() {
    const { history } = this.props;
    history.listen((newLocation, action) => {
      if (action === "REPLACE") {
        if (
          newLocation.pathname !== this.currentPathname ||
          newLocation.search !== this.currentSearch
        ) {
          // Save new location
          this.currentPathname = newLocation.pathname;
          this.currentSearch = newLocation.search;

          // Clone location object and push it to history
          history.replace({
            pathname: newLocation.pathname,
            search: newLocation.search,
          });
        }
      } else {
        // Send user back if they try to navigate back
        history.go(1);
      }
    });
  }

  render() {
    console.log("signupReducer", this.props.signupReducer);
    return (
      <Provider store={store}>
        <Router>
          <Switch>
            <Route
              exact
              path="/"
              render={() => <Redirect to="/loginScreen" />}
            />

            <Route exact path="/loginScreen" component={LoginScreen} />
            <PrivateRoute path="/dashboard">
              <Dashboard />
            </PrivateRoute>
            <PrivateRoute path="/summary">
              <Summary />
            </PrivateRoute>
            <Route path="/dashboard" component={Dashboard} />
            <Route path="/summary" component={Summary} />
          </Switch>
        </Router>
      </Provider>
    );
  }
}

// export default connect(mapStateToProps, null)(App);
export default withRouter(App);
