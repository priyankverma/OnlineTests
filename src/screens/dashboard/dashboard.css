.carousel {
  display: flex;
  flex: 0.7;
  /* height: 75%; */
  box-shadow: inset;
  background-color: white;
  width: 1000px;
}

.buttonView {
  display: flex;
  flex: 0.5;
  align-items: center;
  justify-content: center;
  margin-top: 5%;
}

.bottomButtonView {
  display: flex;
  flex: 0.5;
  justify-content: center;
  background-color: green;
}

.typeQuestion {
  display: flex;
  flex: 1;
  justify-content: flex-end;
  margin-right: 4.5%;
  margin-top: -1.5%;
}
