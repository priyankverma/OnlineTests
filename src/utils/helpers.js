const deviceWidth = {
  mobileS: "250px", // 320 is the standard
  tablet: "768px",
  desktop: "980px",
};

export const device = {
  mobileS: `(min-width: ${deviceWidth.mobileS})`,
  tablet: `(min-width: ${deviceWidth.tablet})`,
  desktop: `(min-width: ${deviceWidth.desktop})`,
};
