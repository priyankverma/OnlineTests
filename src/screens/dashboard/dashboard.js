/**
 * The dashboard component, containing the questions and the answer as option,
 * user can simply selected the options depending on the type of the question ('Single', 'Multiple'),
 * a summary of the total number of questions attempted, the time left and the history of what option is selected for which question.
 *
 * The input to this component is the questionSet, coming from the reducers, and the userInformation,
 * populating from the reducers.
 *
 * User navigates to this page after loggin in, agrreing to the terms and conditions, reading the instructions and
 * selecting the level of questions.
 *
 * After the completion of the timer or on submission by the user, the nesxt screen is Summary page.
 */

import React, { useEffect, useRef, useState } from "react";
import { useDispatch, useSelector } from "react-redux";

import {
  Container,
  Wrapper,
  ContentWrap,
  RowWrap,
  RowWrapQuestions,
  QuestionWrap,
  QuestionsDiv,
  AnswersDiv,
  Box1,
  Box2,
  NextButton,
  StyledCarousal,
  TopButtonDiv,
  BottomButtonView,
} from "./dashboardStyles";
import Header from "../../commonComponents/header/header";
import { Popconfirm } from "antd";
import useWindowDimensions from "./../../commonComponents/customhooks/useWindowDimensions";
import usePersistedState from "./../../commonComponents/customhooks/usePersistedState";

import CircularBar from "../../commonComponents/progressBar/progressBar";
import "./dashboard.css";
import { useHistory, withRouter } from "react-router-dom";
import { registerAction } from "../../redux/actions/signupAction";
window.history.forward(true);
let executeOnce = true;
export const Dashboard = (props) => {
  const slider = useRef();
  const history = useHistory();
  const dispatch = useDispatch();
  const reducerData = useSelector((state) => state);

  const { signupReducer } = reducerData;
  const { data, questionSet } = signupReducer;
  const userData = data;
  const [questionsState, setQuestionsState] = usePersistedState(
    "questionSet",
    questionSet
  );
  const [selectedAnswer, setSelectedAnswer] = useState(null);
  const [questionSelected, setQuestionSelected] = useState(0);
  const { height, width } = useWindowDimensions();
  const [attemptedCount, setAttemptedCount] = usePersistedState(
    "attemptedcount",
    0
  );
  const [executeNow, setExecuteNow] = useState(true);
  const [visible, setVisible] = useState(false);
  const [confirmLoading, setConfirmLoading] = useState(false);
  const [saveTimerVal, setSaveTimerVal] = useState(120);
  let stickyValue = window.localStorage.getItem("TimerValue");
  useEffect(() => {
    let userData = localStorage.getItem("userData");
    let stringifiedUserData = userData !== null ? JSON.parse(userData) : null;
    console.log("stringifiedUserData", stringifiedUserData);
    stringifiedUserData &&
      dispatch(
        registerAction({
          username: stringifiedUserData.username,
          password: stringifiedUserData.password,
          complexityLevel: stringifiedUserData.complexityLevel,
        })
      );
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);
  /**
   * controls the visiblity of the pop up, visible after user taps submit
   */
  const showPopconfirm = () => {
    setVisible(true);
  };
  /**
   * processes the count, makes a navigation action to submit page after user confirms submitting and deactivates the countdown
   */
  const handleOk = () => {
    setConfirmLoading(true);
    questionsState.attemptedCount = attemptedCount;

    setTimeout(() => {
      setVisible(false);
      setConfirmLoading(false);

      handleSaveToPC(questionsState);
      history.replace({
        pathname: "/summary",
        state: { questionSet: questionsState, userData },
      });
    }, 2000);
  };
  /**
   * A function that exports the question set along with the option, answers data and user Response after the submission of test by the user
   * @param {object} jsonData - the data that needs to be exported as the json file after user reaches the summary page
   */
  const handleSaveToPC = (jsonData) => {
    const fileData = JSON.stringify(jsonData);
    const blob = new Blob([fileData], { type: "text/plain" });
    const url = URL.createObjectURL(blob);
    const link = document.createElement("a");
    link.download = data.username + "_Report.json";
    link.href = url;
    link.click();
  };

  /**
   * controls the visiblity of the pop up, visible after user taps submit
   */
  const handleCancel = () => {
    setVisible(false);
  };

  /**
   * A function, executed when user selects the answers from the different options available for a question,
   * handles different cases on the basis of type['Multiple', 'Single']
   * @param {object} questions - current mapped question object from the questionsArray
   * @param {number} questionIndex  - index of the mapped question
   * @param {object} answers - current mapped answers from the current question array
   * @param {number} answersIndex - selected answer index
   */
  const answerClicked = (questions, questionIndex, answers, answersIndex) => {
    if (
      questions.userResponse ||
      (questions.multipleSelected && questions.multipleSelected.length !== 0)
    ) {
      // console.log.bind(console) does nothing if the user has already attempted any question
    } else {
      setAttemptedCount(attemptedCount + 1); // increases the attempted count by 1
    }
    if (questions.type === "multiple") {
      let array = questions.multipleSelected;
      let tempIndex = array.indexOf(answers.option);

      if (tempIndex !== -1 && array.length > 1) {
        // unselecs the option if tapped again in multiple choice question
        array.splice(tempIndex, 1);
      } else {
        questions.multipleSelected = [
          answers.option,
          ...questions.multipleSelected,
        ]; // pushes the index of the answers selected
      }
    } else {
      if (questionsState) {
        questionsState[questionIndex].userResponse = {
          answerSubmitted: answers,
          validAnswer: answers.correctAnswer,
        }; // records the user response in the main question file with the answer selected by user and the status to it, whether correct or not
      }
    }

    setSelectedAnswer(answersIndex);
  };

  return (
    <Container height={height}>
      <Header
        testid="headerComponent"
        userInfo={userData ? userData.username : null}
      />
      <Wrapper>
        <ContentWrap>
          <RowWrapQuestions flex={0.2}>
            {questionsState &&
              questionsState.map((questions, questionIndex) => {
                return (
                  <QuestionWrap
                    key={questionIndex + new Date()}
                    answered={
                      questions.userResponse ||
                      (questions.multipleSelected &&
                        questions.multipleSelected.length !== 0)
                    }
                    leftQuestionSelected={questionSelected === questionIndex}
                    onClick={() => {
                      /**
                       * controls the index jumping from the left component visible only on desktop and tablets
                       */
                      setExecuteNow(false);
                      setSelectedAnswer(null);
                      setTimeout(() => {
                        setQuestionSelected(questionIndex);
                      }, 100);

                      slider.current.goTo(questionIndex, true);
                    }}
                  >
                    Q{questionIndex + 1}. {questions.que}
                  </QuestionWrap>
                );
              })}
          </RowWrapQuestions>
          <RowWrap flex={0.8} backgroundColor={"pink"}>
            <Box1>
              {questionsState && (
                <CircularBar
                  title="Questions Attempted"
                  progressPercent={attemptedCount * 10}
                  bottomText={attemptedCount + "/" + questionsState.length}
                  width={height * 0.15}
                  percent={attemptedCount * 10}
                />
              )}

              <CircularBar
                title="Time Left"
                countDownTimeInSeconds={
                  stickyValue !== null ? JSON.parse(stickyValue) : saveTimerVal
                }
                width={height * 0.15}
                timeOver={() => {
                  if (executeOnce) {
                    executeOnce = false;
                    questionsState.attemptedCount = attemptedCount;

                    handleSaveToPC(questionsState);
                    history.replace({
                      pathname: "/summary",
                      state: { questionSet: questionsState, userData },
                    });
                  }
                }}
                transferVal={(val) => {
                  window.localStorage.setItem(
                    "TimerValue",
                    JSON.stringify(val)
                  );
                  setSaveTimerVal(val);
                }}
              />
            </Box1>
            <Box2>
              <StyledCarousal
                swipe={false}
                ref={(ref) => {
                  slider.current = ref;
                }}
                effect="fade"
                dots={false}
                infinite={false}
                height={height * 0.45}
                width={width}
                beforeChange={(from, to) => {
                  if (executeNow) {
                    if (from > to) {
                      setQuestionSelected(questionSelected - 1);
                    } else {
                      setQuestionSelected(questionSelected + 1);
                    }
                  }
                }}
              >
                {questionsState &&
                  questionsState.map((questions, questionIndex) => {
                    return (
                      <div key={new Date() + questionIndex}>
                        <QuestionsDiv>
                          Q{questionIndex + 1}. {questions.que}
                        </QuestionsDiv>
                        <div className="typeQuestion">
                          [
                          {questions.type === "single"
                            ? "Single Choice"
                            : "Multiple Choice"}
                          ]
                        </div>
                        {questions.answersArr.map((answers, answersIndex) => {
                          return (
                            <AnswersDiv
                              key={new Date() + answersIndex}
                              selected={
                                questions.type === "multiple"
                                  ? questions.multipleSelected &&
                                    questions.multipleSelected.includes(
                                      answers.option
                                    )
                                  : (questions.userResponse &&
                                      questions.userResponse.answerSubmitted
                                        .option === answers.option) ||
                                    selectedAnswer === answersIndex
                              }
                              onClick={() => {
                                setQuestionsState(questionsState);

                                answerClicked(
                                  questions,
                                  questionIndex,
                                  answers,
                                  answersIndex
                                );
                              }}
                            >
                              {(answersIndex + 10).toString(36).toLowerCase()}
                              ). {answers.option}
                            </AnswersDiv>
                          );
                        })}
                      </div>
                    );
                  })}
              </StyledCarousal>

              <TopButtonDiv>
                <NextButton
                  disabled={questionSelected < 1 ? true : false}
                  enabled={questionSelected < 1 ? false : true}
                  paddingRight="5%"
                  onClick={() => {
                    /**
                     * mantains previous button disabled and activated state on the basis of current index
                     */
                    if (questionSelected >= 1) {
                      setExecuteNow(true);
                      setTimeout(() => {
                        slider.current.prev();
                      }, 200);
                      setSelectedAnswer(null);
                    }
                  }}
                  {...props}
                >
                  <b>Previous</b>
                </NextButton>

                <NextButton
                  enabled={
                    questionSelected === questionsState &&
                    questionsState.length - 1
                      ? false
                      : true
                  }
                  onClick={() => {
                    /**
                     * mantains next button disabled and activated state on the basis of current index
                     */
                    if (
                      questionSelected !== questionsState &&
                      questionsState.length - 1
                    ) {
                      setExecuteNow(true);
                      setTimeout(() => {
                        slider.current.next();
                      }, 200);
                      setSelectedAnswer(null);
                    }
                  }}
                  {...props}
                >
                  <b>Next</b>
                </NextButton>
              </TopButtonDiv>
              <BottomButtonView>
                <Popconfirm
                  title="Once Submitted, you won't be able to change answers."
                  visible={visible}
                  onConfirm={handleOk}
                  okButtonProps={{ loading: confirmLoading }}
                  onCancel={handleCancel}
                >
                  <NextButton
                    enabled
                    scheme={"orange"}
                    onClick={() => showPopconfirm()}
                  >
                    <b>Submit</b>
                  </NextButton>
                </Popconfirm>
              </BottomButtonView>
            </Box2>
          </RowWrap>
          {/* <RowWrap flex={0.3} backgroundColor={"violet"}>
            ddddd
          </RowWrap> */}
        </ContentWrap>
      </Wrapper>
    </Container>
  );
};

export default withRouter(Dashboard);
