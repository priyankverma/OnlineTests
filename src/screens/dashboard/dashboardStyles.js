import styled from "styled-components";
import { device } from "./../../utils/helpers";
import { Carousel } from "antd";
export const Container = styled.div`
  display: flex;
  flex: 1;
  flex-direction: column;
  background-color: white;
  overflow-y: hidden;
  height: ${(props) => props.height + "px"};
`;

export const Wrapper = styled.div`
  display: flex;
  flex: 1;
  /* overflow-x: hidden; */
`;

export const ContentWrap = styled.div`
  display: flex;
  flex: 1;
  flex-direction: row;
  height: ${(props) => props.height + "px"};
`;

export const RowWrap = styled.div`
  display: flex;
  flex: ${(props) => props.flex};
  background-color: ${(props) => props.backgroundColor};
  height: 100%;
  flex-direction: column;
  /* overflow-x: hidden; */
  @media ${device.mobileS} {
    flex: 1;
  }
  @media ${device.tablet} {
    flex: ${(props) => props.flex};
  }
  @media ${device.desktop} {
    flex: ${(props) => props.flex};
  }
`;

export const RowWrapQuestions = styled.div`
  display: flex;
  flex: ${(props) => props.flex};
  background-color: ${(props) => props.backgroundColor};
  height: 100%;
  flex-direction: column;
  overflow-y: scroll;
  /* justify-content: space-evenly; */
  border-right: 2px solid #dadada;
  box-shadow: 0px 0px 15px 5px grey;
  border-radius: 5px;
  width: 100%;
  padding: 1%;

  @media ${device.mobileS} {
    display: none;
    visibility: hidden;
  }
  @media ${device.tablet} {
    display: flex;
    visibility: visible;
    flex: ${(props) => props.flex};
  }
  @media ${device.desktop} {
    display: flex;
    visibility: visible;
    flex: ${(props) => props.flex};
  }
`;

export const QuestionWrap = styled.div`
  display: flex;
  /* flex: 1; */
  height: 5%;
  margin-top: 3%;
  overflow-x: hidden;
  text-align: left;
  background-image: ${(props) =>
    props.answered
      ? "linear-gradient(to right, #3ca55c, #b5ac49)"
      : "linear-gradient(to right, lightgrey, lightgrey)"};
  align-items: flex-start;
  justify-content: flex-start;
  border-radius: 5px;
  padding: 2%;
  border: ${(props) =>
    props.leftQuestionSelected ? "3px solid #3a3985" : "1px solid #3a3985"};
  cursor: pointer;
  color: ${(props) => (props.answered ? "white" : "black")};
`;

export const QuestionsDiv = styled.div`
  margin-top: 2%;
  margin-bottom: 2%;
  background-color: lightgrey;
  padding: 10px;
  margin-left: 4%;
  margin-right: 4%;
  border-radius: 5px;
  align-items: flex-start;
  text-align: left;
  border: 1px solid #3a3985;
`;

export const AnswersDiv = styled.div`
  margin-top: 2%;
  margin-bottom: 2%;
  overflow-x: hidden;
  background-image: ${(props) =>
    props.selected
      ? "linear-gradient( to right top, #fab823, #f9a315, #f68d0c, #f3770d,#ee5f13)"
      : "linear-gradient( to right top, lightgrey, lightgrey)"};
  padding: 10px;
  margin-left: 7%;
  border-radius: 5px;
  text-align: left;
  margin-right: 4%;
  color: ${(props) => (props.selected ? "white" : "black")};
  cursor: pointer;
  /* height: 5%; */
`;
export const Box1 = styled.div`
  display: flex;
  flex: 0.3;
  box-shadow: 0px 0px 5px 2px grey;
  border-radius: 5px;
  background-color: white;
  margin: 4px;
  justify-content: space-around;
  overflow-x: hidden;
  @media ${device.mobileS} {
    flex: 0.3;
  }
  @media ${device.tablet} {
    flex: 0.4;
  }
  @media ${device.desktop} {
    flex: 0.4;
  }
`;
export const Box2 = styled.div`
  display: flex;
  flex: 0.7;
  /* height: 100%; */
  flex-direction: column;
  box-shadow: 0px 0px 5px 2px grey;
  border-radius: 5px;
  background-color: white;
  margin: 1%;
  overflow-x: hidden;
  /* position: absolute; */

  flex-direction: column;
  b {
    color: whitesmoke;
  }
  @media ${device.mobileS} {
    flex: 0.7;
  }
  @media ${device.tablet} {
    flex: 0.6;
  }
  @media ${device.desktop} {
    flex: 0.6;
  }
`;

export const NextButton = styled.button`
  /* flex: 0.1; */
  opacity: ${(props) => (props.enabled ? 1 : 0.6)};
  display: flex;
  background-image: ${(props) =>
    props.scheme === "orange"
      ? "linear-gradient( to right top, #fab823, #f9a315, #f68d0c, #f3770d,#ee5f13)"
      : "linear-gradient(to right top, #3499ff, #3a3985)"};
  height: ${(props) => (props.height ? props.height + "px" : "30%")};
  width: 8%;
  border-radius: 5px;
  align-items: center;
  justify-content: center;
  cursor: pointer;
  margin-right: ${(props) => (props.paddingRight ? props.paddingRight : 0)};
  /* padding: 3px 10px 3px 10px; */

  @media ${device.mobileS} {
    width: 24%;
    height: 50%;
  }
  @media ${device.tablet} {
    height: ${(props) => (props.height ? props.height + "px" : "50%")};
    width: 16%;
  }
  @media ${device.desktop} {
    height: ${(props) => (props.height ? props.height + "px" : "70%")};
    width: 15%;
  }
`;

export const StyledCarousal = styled(Carousel)`
  display: flex;
  flex: 0.5;
  background-color: white;
  border-radius: 5px;
  height: ${(props) => props.height + "px"};
  width: ${(props) => props.width + "px"};
  overflow-y: hidden;
  @media ${device.mobileS} {
    width: ${(props) => props.width + "px"};
    height: ${(props) => props.height + "px"};
  }
  @media ${device.tablet} {
    width: ${(props) => props.width * 0.75 + "px"};
  }
  @media ${device.desktop} {
    width: ${(props) => props.width * 0.75 + "px"};
  }
`;

export const TopButtonDiv = styled.div`
  display: flex;
  flex: 0.5;
  align-items: center;
  justify-content: center;
  background-color: white;
  @media ${device.mobileS} {
    margin-top: 5%;
  }
  @media ${device.tablet} {
    margin-top: 0%;
  }
  @media ${device.desktop} {
    margin-top: 0%;
  }
`;
export const BottomButtonView = styled.div`
  display: flex;
  flex: 0.5;
  justify-content: center;
  background-color: white;
`;
