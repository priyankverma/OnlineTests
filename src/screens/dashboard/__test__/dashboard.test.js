import { render } from "@testing-library/react";
import * as React from "react";
import { Provider } from "react-redux";
import { store } from "./../../../App";
import { Dashboard } from "../dashboard";
import { shallow } from "./../../../utils/enzyme";
import { NextButton } from "../dashboardStyles";
import renderer from "react-test-renderer";

describe("Login Component", () => {
  let DashComponent = (
    <Provider store={store}>
      <Dashboard />
    </Provider>
  );

  it("renders header correctly", () => {
    const { getByTestId } = render(DashComponent);
    expect(getByTestId("headerComponent")).toBeVisible();
  });

  it("previous button is initially disabled", () => {
    const { getByText } = render(DashComponent);
    expect(getByText(/Previous/i).closest("button")).toBeDisabled();
  });
  it("next button is initially enabled", () => {
    const { getByText } = render(DashComponent);
    expect(getByText(/Next/i).closest("button")).toBeEnabled();
  });
  it("submit button is always enabled", () => {
    const { getByText } = render(DashComponent);
    expect(getByText(/Submit/i).closest("button")).toBeEnabled();
  });
  it("next button is disabled on swiping to last question", () => {
    const { getByText } = render(DashComponent);
    expect(getByText(/Next/i).closest("button")).toBeEnabled();
  });

  it("custom button click event works fine", () => {
    const mockCallBack = jest.fn();

    const button = shallow(<NextButton onClick={mockCallBack}>Ok!</NextButton>);
    button.find("button").simulate("click");
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });

  it("props are not available initially", () => {
    const wrapper = shallow(DashComponent);
    expect(wrapper.prop("questionSet")).toBeUndefined();
  });
  it("matches snapshot perfectly", () => {
    const tree = renderer.create(DashComponent).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
