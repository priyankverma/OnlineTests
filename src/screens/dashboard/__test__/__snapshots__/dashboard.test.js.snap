// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component matches snapshot perfectly 1`] = `
<div
  className="sc-bdvvaa fjwAeu"
  height={768}
>
  <div
    className="sc-egiSv coTDwk"
    data-testid="headerComponent"
  >
    <div
      className="sc-bqiQRQ gfBxLs"
      onClick={[Function]}
    >
      <img
        alt={Object {}}
        src="logo.png"
      />
    </div>
    <div
      className="sc-bqiQRQ idjuKp"
      onClick={[Function]}
    >
      <b>
        Thu, May 13, 2021
      </b>
    </div>
    <div
      className="sc-bqiQRQ cwUtdq"
      onClick={[Function]}
    >
      <b>
        <span
          aria-label="user"
          className="anticon anticon-user"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="user"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
            />
          </svg>
        </span>
         
         null
      </b>
    </div>
  </div>
  <div
    className="sc-gsDJrp iTIzpr"
  >
    <div
      className="sc-dkPtyc vMfYF"
    >
      <div
        className="sc-eCImvq KewmA"
      />
      <div
        className="sc-hKwCoD fFHCMR"
      >
        <div
          className="sc-furvIG frssmD"
        >
          <div
            className="sc-ksdxAp fgsQyf"
          >
            <div
              className="sc-fFehDp exyvYJ"
            >
              <b
                className="sc-hBURRC bZXNwG"
              >
                Questions Attempted
              </b>
              <div
                className="ant-progress ant-progress-circle ant-progress-status-normal ant-progress-default"
              >
                <div
                  className="ant-progress-inner ant-progress-circle-gradient"
                  style={
                    Object {
                      "fontSize": 23.279999999999998,
                      "height": 115.19999999999999,
                      "width": 115.19999999999999,
                    }
                  }
                >
                  <svg
                    className="ant-progress-circle"
                    style={Object {}}
                    viewBox="0 0 100 100"
                  >
                    <defs>
                      <linearGradient
                        id="ant-progress-gradient-11"
                        x1="100%"
                        x2="0%"
                        y1="0%"
                        y2="0%"
                      >
                        <stop
                          offset="0%"
                          stopColor="#108ee9"
                        />
                        <stop
                          offset="100%"
                          stopColor="#87d068"
                        />
                      </linearGradient>
                    </defs>
                    <path
                      className="ant-progress-circle-trail"
                      d="M 50,50 m 0,-47
   a 47,47 0 1 1 0,94
   a 47,47 0 1 1 0,-94"
                      fillOpacity="0"
                      stroke={null}
                      strokeLinecap="round"
                      strokeWidth={6}
                      style={
                        Object {
                          "stroke": null,
                          "strokeDasharray": "295.3097094374406px 295.3097094374406px",
                          "strokeDashoffset": "-0px",
                          "transition": "stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s, opacity .3s ease 0s",
                        }
                      }
                    />
                    <path
                      className="ant-progress-circle-path"
                      d="M 50,50 m 0,-47
   a 47,47 0 1 1 0,94
   a 47,47 0 1 1 0,-94"
                      fillOpacity="0"
                      opacity={0}
                      stroke="url(#ant-progress-gradient-11)"
                      strokeLinecap="round"
                      strokeWidth={6}
                      style={
                        Object {
                          "stroke": Object {
                            "0%": "#108ee9",
                            "100%": "#87d068",
                          },
                          "strokeDasharray": "0px 295.3097094374406px",
                          "strokeDashoffset": "-0px",
                          "transition": "stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s, opacity .3s ease 0s",
                        }
                      }
                    />
                  </svg>
                </div>
              </div>
              <b
                className="sc-fotPbf jeJztL"
              >
                0/0
              </b>
            </div>
          </div>
          <div
            className="sc-ksdxAp fgsQyf"
          >
            <div
              className="sc-fFehDp exyvYJ"
            >
              <b
                className="sc-hBURRC bZXNwG"
              >
                Time Left
              </b>
              <div
                className="ant-progress ant-progress-circle ant-progress-status-success ant-progress-default"
              >
                <div
                  className="ant-progress-inner ant-progress-circle-gradient"
                  style={
                    Object {
                      "fontSize": 23.279999999999998,
                      "height": 115.19999999999999,
                      "width": 115.19999999999999,
                    }
                  }
                >
                  <svg
                    className="ant-progress-circle"
                    style={Object {}}
                    viewBox="0 0 100 100"
                  >
                    <defs>
                      <linearGradient
                        id="ant-progress-gradient-12"
                        x1="100%"
                        x2="0%"
                        y1="0%"
                        y2="0%"
                      >
                        <stop
                          offset="0%"
                          stopColor="#108ee9"
                        />
                        <stop
                          offset="100%"
                          stopColor="#87d068"
                        />
                      </linearGradient>
                    </defs>
                    <path
                      className="ant-progress-circle-trail"
                      d="M 50,50 m 0,-47
   a 47,47 0 1 1 0,94
   a 47,47 0 1 1 0,-94"
                      fillOpacity="0"
                      stroke={null}
                      strokeLinecap="round"
                      strokeWidth={6}
                      style={
                        Object {
                          "stroke": null,
                          "strokeDasharray": "295.3097094374406px 295.3097094374406px",
                          "strokeDashoffset": "-0px",
                          "transition": "stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s, opacity .3s ease 0s",
                        }
                      }
                    />
                    <path
                      className="ant-progress-circle-path"
                      d="M 50,50 m 0,-47
   a 47,47 0 1 1 0,94
   a 47,47 0 1 1 0,-94"
                      fillOpacity="0"
                      opacity={1}
                      stroke="url(#ant-progress-gradient-12)"
                      strokeLinecap="round"
                      strokeWidth={6}
                      style={
                        Object {
                          "stroke": Object {
                            "0%": "#108ee9",
                            "100%": "#87d068",
                          },
                          "strokeDasharray": "295.3097094374406px 295.3097094374406px",
                          "strokeDashoffset": "-0px",
                          "transition": "stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s, opacity .3s ease 0s",
                        }
                      }
                    />
                  </svg>
                </div>
              </div>
              <b
                className="sc-fotPbf jeJztL"
              >
                01:54
              </b>
            </div>
          </div>
        </div>
        <div
          className="sc-pVTma fVBZJG"
        >
          <div
            className="ant-carousel"
          >
            <div
              className="slick-slider sc-kDThTU dHnoAo slick-initialized"
            >
              
              <div
                className="slick-list"
              >
                <div
                  className="slick-track"
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onMouseOver={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                />
              </div>
              
              
            </div>
          </div>
          <div
            className="sc-iqsfdx cEOmMR"
          >
            <button
              className="sc-jrQzUz kqiGCa"
              disabled={true}
              onClick={[Function]}
            >
              <b>
                Previous
              </b>
            </button>
            <button
              className="sc-jrQzUz jjrVVn"
              onClick={[Function]}
            >
              <b>
                Next
              </b>
            </button>
          </div>
          <div
            className="sc-crHlIS gyuMRu"
          >
            <button
              className="sc-jrQzUz dGLqnk"
              onClick={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            >
              <b>
                Submit
              </b>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
